hub/api/objectview.py

```diff
@@ -0,0 +1,373 @@
+from hub.api.datasetview import DatasetView
+from hub.schema import Sequence, Tensor, SchemaDict, Primitive
+from hub.api.dataset_utils import slice_extract_info, slice_split, str_to_int
+
+# from hub.exceptions import NoneValueException
+import collections.abc as abc
+import hub.api as api
+
+
+class ObjectView:
+    def __init__(
+        self,
+        dataset,
+        subpath=None,
+        slice_list=None,
+        nums=[],
+        offsets=[],
+        squeeze_dims=[],
+        inner_schema_obj=None,
+        new=True,
+    ):
+        self.dataset = dataset
+        self.schema = (
+            dataset.schema.dict_
+            if not isinstance(dataset, DatasetView)
+            else dataset.dataset.schema.dict_
+        )
+        # assert isinstance(self.schema, dict)
+        self.subpath = subpath
+
+        self.nums = nums
+        self.offsets = offsets
+        self.squeeze_dims = squeeze_dims
+
+        self.inner_schema_obj = inner_schema_obj
+
+        if new:
+            # Creating new obj
+            if self.subpath:
+                (
+                    self.inner_schema_obj,
+                    self.nums,
+                    self.offsets,
+                    self.squeeze_dims,
+                ) = self.process_path(
+                    self.subpath,
+                    self.inner_schema_obj,
+                    self.nums.copy(),
+                    self.offsets.copy(),
+                    self.squeeze_dims.copy(),
+                )
+            # Check if dataset view needs to be made
+            if slice_list and len(slice_list) >= 1:
+                num, ofs = slice_extract_info(slice_list[0], dataset.shape[0])
+                self.dataset = DatasetView(
+                    dataset, num, ofs, isinstance(slice_list[0], int)
+                )
+
+            if slice_list and len(slice_list) > 1:
+                slice_list = slice_list[1:]
+                if len(slice_list) > len(self.nums):
+                    raise IndexError("Too many indices")
+                for i, it in enumerate(slice_list):
+                    num, ofs = slice_extract_info(it, self.nums[i])
+                    self.nums[i] = num
+                    self.offsets[i] += ofs
+                    self.squeeze_dims[i] = num == 1
+
+    def num_process(self, schema_obj, nums, offsets, squeeze_dims):
+        if isinstance(schema_obj, SchemaDict):
+            return
+        # if isinstance(schema_obj.max_shape, int):
```

Comment:
Clean out the commented code throughout.

Reply:
Done